tensorboard/webapp/widgets/line_chart_v2/line_chart_component.ts

```diff
@@ -226,10 +229,22 @@ export class LineChartComponent
       this.useDarkModeUpdated = true;
     }
 
-    if (this.scaleUpdated) {
+    // Do not set isViewBoxOverridden to false until updated userViewBox has been reflected on the lineChart.
+    if (this.scaleUpdated && !this.userViewBoxUpdated) {
       this.setIsViewBoxOverridden(false);
     }
 
+    if (changes['userViewBox']) {
+      this.setIsViewBoxOverridden(false);
+      this.isViewBoxChanged = true;
```

Comment:
this.isViewBoxChanged has a complicated calculation on L248. It seems like whatever value you set here will just immediately be overriden by whatever is set on L248. Am I missing something?